cupy/core/_cub_reduction.pyx

```diff
@@ -542,21 +530,23 @@ def _get_cub_optimized_params(
             post_map_expr, reduce_type, stream, params, cub_params)
 
     def suggest_func(trial):
-        block_size_log = trial.suggest_int('block_size_log', 5, 9)
+        block_size_log = trial.suggest_int('block_size_log', 5, 10)
         block_size = 2 ** block_size_log
         items_per_thread = trial.suggest_int(
             'items_per_thread', 2, 32, step=2)
 
         trial.set_user_attr('block_size', block_size)
         return block_size, items_per_thread
 
+    # CUDA_ERROR_LAUNCH_OUT_OF_RESOURCES is a possible error
+    from cupy.cuda import driver
```

Comment:
Why not import `driver` at the top of this file?

Reply:
This way we don't import until the optimizer is on? I can change that.

Reply:
`cupy.cuda.driver` will be imported anywhere from CuPy codebase, so we don't need to take care of it.

Reply:
Thanks, changed in aecb521.